tool/yarn

```diff
@@ -0,0 +1,71 @@
+#!/bin/sh
+#
+# This script acts like the "yarn" command, but uses Tailscale's
+# currently-desired version, downloading it (and node) first if necessary.
+
+set -eu
+
+NODE_DIR="$HOME/.cache/tailscale-node"
+YARN_DIR="$HOME/.cache/tailscale-yarn"
+
+install_node() {
+    read -r REV < "$(dirname "$0")/node.rev"
+    # works for "linux" and "darwin"
+    OS=$(uname -s | tr A-Z a-z)
+    ARCH=$(uname -m)
+
+    if [ "$ARCH" = "x86_64" ]; then
+        ARCH="x64"
+    fi
+    if [ "$ARCH" = "aarch64" ]; then
+        ARCH="arm64"
+    fi
+    NODE_URL="https://nodejs.org/dist/v${REV}/node-v${REV}-${OS}-${ARCH}.tar.gz"
+    install_tool "node" $REV $NODE_DIR $NODE_URL
+}
+
+install_yarn() {
+    read -r REV < "$(dirname "$0")/yarn.rev"
```

Comment:
Add the equivalent of SKIP from `./tool/go`? Something that would use whatever is already available in the environment. Otherwise it'll break on nixos

Reply:
Done. Also added a way to specify a custom yarn path when building, so that we can use `tool/yarn` when doing deploys from corp (that wrapper has nixos-specific logic, which I don't want to replicate).